modular_skyrat/modules/xenos_skyrat_redo/code/xeno_types/queen.dm

```diff
@@ -5,6 +5,7 @@
 	desc = "A hulking beast of an alien, for some reason this one seems more important than the others, you should probably quit staring at it and do something."
 	caste = "queen"
 	maxHealth = 500
+	status_flags = NONE //can't shove or KO the queen, kiddo.
```

Comment:
This is not the fix. Please actually test your PRs.

You want to override the `get_shove_flags` proc and just return. If you want it to be the same as TG, look at their alien shove proc.

For stuns, use `add_stun_absorption` and use src as the source.

Reply:
Setting status flags to NONE does in fact make it immune to KO's, but pushing still occurs.

Reply:
One of the bitfields was CAN_PUSH, and thus I believed this to stop shoves as well.
![image](https://github.com/user-attachments/assets/0c964d38-aebd-4f6e-ab8d-1aab7bcb552c)

Reply:
Mech shoving uses step_away which is a byond builtin, it's unlikely that overriding get_shove_flags will help but I'll test it

Reply:
Yep, get_shove_flags is used for right click shoves, which queen is already immune to

Reply:
@RimiNosha xenos were already immune to stuns, but mech punches and human punches cause unconciousness, which is the issue here

Reply:
What the hell, I think my VSC search was bugged, then, cause all I saw was it being set to none and no uses. My bad.

Reply:
this is a test